torchaudio/csrc/sox.cpp

```diff
@@ -2,12 +2,80 @@
 
 #include <algorithm>
 #include <cstdint>
+#include <memory>
 #include <stdexcept>
 #include <vector>
 
+#ifdef _WIN32
+// Additional headers for implementation for mkstemp
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+
+#include <fcntl.h>
+#include <io.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <Windows.h>
+
+#include <random>
+#include <string>
+#include <vector>
+#endif
+
 namespace torch {
 namespace audio {
 namespace {
+#ifdef _WIN32
+int mkstemp(char* tmpl) {
```

Comment:
Is this your implementation, or taken from somewhere?

Reply:
It comes from my PR: https://github.com/pytorch/pytorch/pull/25578/files#diff-dc1022da8c9adb512abfa2caa114008dR30.